keras_cv/layers/preprocessing/random_gaussian_blur.py

```diff
@@ -94,12 +94,13 @@ def augment_segmentation_mask(self, segmentation_mask, transformation, **kwargs)
 
     @staticmethod
     def get_kernel(factor, filter_size):
+
         x = tf.cast(
-            tf.range(-filter_size // 2 + 1, filter_size // 2 + 1), dtype=tf.float32
-        )
-        blur_filter = tf.exp(
-            -tf.pow(x, 2.0) / (2.0 * tf.pow(tf.cast(factor, dtype=tf.float32), 2.0))
+            tf.range(-filter_size // 2 + 1, filter_size // 2 + 1), dtype=factor.dtype
         )
+        # TODO: are we ok with running this in FP16?
```

Comment:
I would expect this to be okay in fp16, and unless there's some good evidence that this is bad, I don't think we should special case this.

Reply:
exponential computation might be a problem in fp16, how do we verify if it's ok to do this?

Reply:
In PyTorch vision they have test coverage for fp16:

https://github.com/pytorch/vision/blob/main/test/test_functional_tensor.py#L1097-L1146

Reply:
I inspected the output pixel values of `RandomGaussianBlur`. If we calculate the kernel in FP16 the results are quite off, compared to FP32 version (e.g. given value range of 0-255, FP32 will output 240 and FP16 will output 160). 

I will keep kernel calculation in FP32, as this fixes the issue. The blurring is still done in FP16 (after casting the kernel to adequate dtype).

Reply:
Moved (only) kernel calculation fo FP32.